app/src/pages/playground/schemas.ts

```diff
@@ -156,10 +156,23 @@ const stringToInvocationParametersSchema = z
           ),
         }))
         // reparse the object to ensure the mapped keys are also validated
-        .transform(invocationParameterSchema.parse)
         .parse(json)
     );
   })
+  .transform((v, ctx) => {
+    const result = invocationParameterSchema.safeParse(v);
+    if (!result.success) {
+      // bubble errors up to the original schema
+      result.error.issues.forEach((issue) => {
+        ctx.addIssue(issue);
+      });
+      // https://zod.dev/?id=validating-during-transform
```

Comment:
very nice, i figured there was some way to do this but didn't see this, this is awesome